peyvandak/Heidari/url_shortener_final/client.py

```diff
@@ -0,0 +1,113 @@
+import requests
+import random
+import numpy as np
+import threading
+import concurrent.futures
+import time
+import traceback
+import json
+
+def login(username, password, timeout=None):
+    data = {'username': username, 'password': password}
+    response = requests.post('http://localhost:8000/users/login', data=data, timeout=timeout)
+    return response
+
+def logout(timeout=None):
+    response = requests.get('http://localhost:8000/users/logout', timeout=timeout)
+    return response
+
+def register(username, password1, password2, timeout=None):
+    data = {'username': username, 'password1': password1, 'password2': password2}
+    response = requests.post('http://localhost:8000/users/register', data=data, timeout=timeout)
+    return response
+
+def send_long_url(long_url, timeout=None):
+    data = {'long_url': long_url}
+    response = requests.post('http://localhost:8000/create', data=data, timeout=timeout)
+    return response
+
+def send_short_url(short_url, timeout=None):
+    response = requests.get(short_url, timeout=timeout)
+    return response
+
+def get_my_urls(timeout=None):
+    response = requests.get('http://localhost:8000/users/get-my-urls', timeout=timeout)
+    return response
+
+def get_whoami(timeout=None):
+    response = requests.get('http://localhost:8000/users/whoami', timeout=timeout)
+    return response
+
+def get_status(stop_event, running_time):
+    start_time = time.time()
+    time.sleep(running_time)
+    stop_event.set()
+    total_reqs = 0
+    total_success = 0
+    for c in clients:
+        total_reqs += c.total_reqs
+        total_success += c.successful_reqs
+    stop_time = time.time()
+    print(f'total requests: {total_reqs}, successful requests: {total_success}')
+    print(f'success rate: {np.round(total_success/total_reqs, 3) * 100} %, time elapsed: {stop_time - start_time}')
+
+
+class Client:
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+        self.total_reqs = 0
+        self.successful_reqs = 0
+
+    def attack(self, stop_event, total_time):
```

Comment:
Is this function used for testing the API? 
Maybe you can choose a better name for this function.